core/domain/classifier_services.py

```diff
@@ -224,3 +226,117 @@ def delete_classifier(classifier_id):
     classifier_data_model = classifier_models.ClassifierDataModel.get(
         classifier_id)
     classifier_data_model.delete()
+
+
+def get_classifier_training_job_from_model(classifier_training_job_model):
+    """Gets a classifier training job domain object from a classifier
+       training job model.
+
+    Args:
+        classifier_training_job_model: Classifier training job instance in
+            datastore.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+    """
+    return classifier_domain.ClassifierTrainingJob(
+        classifier_training_job_model.id,
+        classifier_training_job_model.algorithm_id,
+        classifier_training_job_model.exp_id,
+        classifier_training_job_model.exp_version_when_created,
+        classifier_training_job_model.state_name,
+        classifier_training_job_model.status,
+        classifier_training_job_model.training_data)
+
+def get_classifier_training_job_by_id(job_id):
+    """Gets a classifier training job from a job_id.
+
+    Args:
+        job_id: str. ID of the classifier training job.
+
+    Returns:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Raises:
+        Exception: Entity for class ClassifierTrainingJobModel with id not
+            found.
+    """
+    classifier_training_job_model = (
+        classifier_models.ClassifierTrainingJobModel.get(job_id))
+    classifier_training_job = get_classifier_training_job_from_model(
+        classifier_training_job_model)
+    return classifier_training_job
+
+
+def _create_classifier_training_job(classifier_training_job):
+    """Creates classifier training job model in the datastore given a
+       classifier training job domain object.
+
+    Args:
+        classifier_training_job: Domain object for the classifier training job.
+
+    Returns:
+        job_id: str. ID of the classifier training job.
+    """
+    job_id = classifier_models.ClassifierTrainingJobModel.create(
+        classifier_training_job.algorithm_id, classifier_training_job.exp_id,
+        classifier_training_job.exp_version_when_created,
```

Comment:
Yep, I remember that discussion too -- and I'm fine with whatever you both decide to go with (but I think it's a good idea for both of you to have the same view on it).

In retrospect, I think what I wanted to ask is: would it be fine to just call this "exp_version" instead of "exp_version_when_created"? (My initial question was incorrect /  badly phrased, so apologies for that.) I remember suggesting the latter for the trained classifiers themselves, since they may be valid for a range of versions. But I'm not sure about the point of carrying this terminology over to the training job models since those pretty obviously correspond to a specific version of an exploration.